tests/unit/verticals/secret/output/test_sarif_output.py

```diff
@@ -0,0 +1,291 @@
+import json
+from typing import Any, Dict, TypedDict
+from unittest import mock
+
+import pytest
+from pygitguardian.models import PolicyBreak, ScanResult
+from pytest_voluptuous import S
+from voluptuous import Optional as VOptional
+from voluptuous import validators
+
+from ggshield.core.scan import Commit
+from ggshield.verticals.secret import Result, Results, SecretScanCollection
+from ggshield.verticals.secret.output import SecretSARIFOutputHandler
+from ggshield.verticals.secret.output.secret_sarif_output_handler import SCHEMA_URL
+from tests.unit.conftest import (
+    _MULTI_SECRET_ONE_LINE_FULL_PATCH,
+    _MULTI_SECRET_ONE_LINE_PATCH_SCAN_RESULT,
+    _MULTIPLE_SECRETS_PATCH,
+    _MULTIPLE_SECRETS_SCAN_RESULT,
+    _ONE_LINE_AND_MULTILINE_PATCH,
+    _ONE_LINE_AND_MULTILINE_PATCH_SCAN_RESULT,
+)
+
+
+VERSION_VALIDATOR = validators.Match(r"\d+\.\d+\.\d+")
+
+
+TOOL_SCHEMA = S(
+    {
+        "driver": {
+            "organization": "GitGuardian",
+            "name": "ggshield",
+            "informationUri": str,
+            "version": VERSION_VALIDATOR,
+        },
+        "extensions": [
+            {
+                "name": str,
+                "version": VERSION_VALIDATOR,
+            }
+        ],
+    }
+)
+
+EMPTY_RESULT_SCHEMA = S(
+    {
+        "version": "2.1.0",
+        "$schema": SCHEMA_URL,
+        "runs": [{"tool": TOOL_SCHEMA, "results": []}],
+    }
+)
+
+MIN_1_INT = validators.Range(min=1)
+
+SARIF_PHYSICAL_LOCATION_DICT_SCHEMA = S(
+    {
+        "artifactLocation": {
+            "uri": str,
+        },
+        "region": {
+            "startLine": MIN_1_INT,
+            "startColumn": MIN_1_INT,
+            "endLine": MIN_1_INT,
+            "endColumn": MIN_1_INT,
+        },
+    }
+)
+
+SARIF_RESULT_DICT_SCHEMA = S(
+    {
+        "ruleId": str,
+        "level": "error",
+        "message": {
+            "text": str,
+            "markdown": str,
+        },
+        "locations": [
+            {
+                "physicalLocation": SARIF_PHYSICAL_LOCATION_DICT_SCHEMA,
+            }
+        ],
+        "relatedLocations": [
+            {
+                "id": int,
+                "physicalLocation": SARIF_PHYSICAL_LOCATION_DICT_SCHEMA,
+                "message": {"text": str},
+            }
+        ],
+        "partialFingerprints": {"secret/v1": str},
+        VOptional("hostedViewerUri"): str,
+    }
+)
```

Comment:
I don't know if it is worth the trouble but on a result on `scan path` the [Sarif validator that you linked](https://sarifweb.azurewebsites.net/Validation) points out other rules that we don't follow and means that we could not be imported by Azure DevOps Advanced Security service and GitHub Advanced Security service.